engine/runtime-instrument-common/src/main/java/org/enso/interpreter/service/ExecutionCallbacks.java

```diff
@@ -214,20 +215,29 @@ private FunctionCallInfo functionCallInfoById(UUID nodeId) {
     return calls.get(nodeId);
   }
 
-  private String typeOf(Object value) {
-    String resultType;
+  private String[] typeOf(Object value) {
     if (value instanceof UnresolvedSymbol) {
-      resultType = Constants.UNRESOLVED_SYMBOL;
-    } else {
-      var typeOfNode = TypeOfNode.getUncached();
-      Object typeResult = value == null ? null : typeOfNode.execute(value);
-      if (typeResult instanceof Type t) {
-        resultType = getTypeQualifiedName(t);
-      } else {
-        resultType = null;
+      return new String[] {Constants.UNRESOLVED_SYMBOL};
+    }
+
+    if (value instanceof EnsoMultiValue multiValue) {
```

Comment:
Ideally this should be encapsulated in `TypeOfNode`. Having these extra checks for `EnsoMultiValue` and `UnresolvedSymbol` here will just cause inconsistency in the future.